bool any_collision
bool[] r_finger_tip
bool[] l_finger_tip
